CMake configuration error in CFS build

**Describe the bug**
When building CFS in certain configurations, it creates the "bplib" target in a way based on the project config.  Sometimes this is STATIC or SHARED but in some cases might also be the MODULE type.

Unfortunately, if it does use the MODULE type, then it is no longer linkable with a normal executable.  This is a problem if tests are enabled, now producing this error:

```
CMake Error at /home/joe/code/cfecfs/github/libs/bplib/ut-functional/CMakeLists.txt:26 (target_link_libraries):
  Target "bplib" of type MODULE_LIBRARY may not be linked into another
  target.  One may link only to INTERFACE, OBJECT, STATIC or SHARED
  libraries, or to executables with the ENABLE_EXPORTS property set.
```

**To Reproduce**
Configure BPLIB as a CFS module, and build with ENABLE_UNIT_TESTS=TRUE

**Expected behavior**
Should build cleanly

**System observed on:**
Ubuntu 22.04

**Reporter Info**
Joseph Hickey, Vantage Systems, Inc.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -209,5 +209,10 @@ target_include_directories(bplib PUBLIC
 
 if (BPLIB_ENABLE_UNIT_TESTS)
   add_subdirectory(ut-stubs)
-  add_subdirectory(ut-functional)
+
+  # BPLib Sanity checks are only for standalone builds, sanity checks
+  # for CFE/CFS builds should be part of the BP app, as opposed to BPLib
+  if (NOT IS_CFS_ARCH_BUILD)
+    add_subdirectory(ut-functional)
+  endif (NOT IS_CFS_ARCH_BUILD)
 endif (BPLIB_ENABLE_UNIT_TESTS)

